src/prefect/workers/base.py

```diff
@@ -358,6 +361,30 @@ def __init__(
         self._cancelling_flow_run_ids = set()
         self._scheduled_task_scopes = set()
 
+        self._last_polled_time: pendulum.DateTime = pendulum.now("utc")
+
+    async def check_if_worker_is_polling(self) -> JSONResponse:
+        """
+        If this health check is invoked, and we have not registered a poll
+        in the last 5 minutes, return a 503 so that worker can be
+        restarted by a container orchestrator liveness probe (or similar)
+        """
+        seconds_since_last_poll = (
+            pendulum.now("utc") - self._last_polled_time
+        ).in_seconds()
+
+        if seconds_since_last_poll >= 300:
+            self._logger.error(
+                f"Worker has not polled in the last {seconds_since_last_poll} seconds "
+                "and should be restarted"
+            )
+
+            return JSONResponse(
+                status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
+                content={"message": "Worker may be unresponsive at this time"},
+            )
+        return JSONResponse(status_code=status.HTTP_200_OK, content={"message": "OK"})
```

Comment:
i have some mixed feelings about coupling webserver implementation details with the BaseWorker class, so im open to ideas of how to decouple.  or maybe it doesn't matter in the long run

Reply:
actually, im gonna move the route handler to `server.py`, and keep this to be a straight up boolean check